### DISCUSSION TITLE
Configuring Digital Inputs - Phidget HUB0001

### DISCUSSION THREAD

**Discussion by KT-19903** [artisan-roaster-scope/artisan#1541] [General]
Hi,

I'm basically trying to emulate what is being done here: https://www.bigloose.com/v11/index.php/home/archive/a-physical-1st-crack-button-in-artisan-scope

I've been reading through all the relevant posts I could find on configuring digital inputs with Phidgets. It appears very straightforward, but I'm not getting the expected result. 

I'm using a HUB0001, configured as an extra device to read a simple switch (the last item below):
![Artisan Devices](https://github.com/artisan-roaster-scope/artisan/assets/167718866/221c4446-1918-4d39-a8fc-692a43ee1480)

I have an alarm configured per guidance Makomo provided in another thread that I understand should work... but it seems Artisan is not receiving the digital input correctly. Once everything is turned on, the LCD only reports "u.u" and the alarms never trigger.

![Artisan Alarms](https://github.com/artisan-roaster-scope/artisan/assets/167718866/b35995f1-d687-4e6b-bc97-71c4d98e1517)

![Artisan LCDs](https://github.com/artisan-roaster-scope/artisan/assets/167718866/26e583d8-1d88-4838-b418-fea1d4a9ec76)

![Artisan Messages](https://github.com/artisan-roaster-scope/artisan/assets/167718866/188e62fd-fd18-467f-89a4-507bcb472226)

When I configure the extra device as the analog input rather than the digital input, data is passed through - but a digital input is desired. 

Am I missing something simple? Surely I am... I appreciate your help.

Artisan settings below:

[artisan-settings.TXT](https://github.com/artisan-roaster-scope/artisan/files/15055268/artisan-settings.TXT)

Thanks!

Keith

**Comment by MAKOMO** [comment]
At which ports are your TMP1101 and Phidget IO modules attached? If any module is attached to port 4 or 5, the HUB_45 port cannot be attached.

**Reply by KT-19903** [reply]
Hi Marko,

The layout is like this:

HUB0001:
>Port 0: TMP1101_1
>Port 1: TMP1200_0
>Port 2: Open
>Port 3: Open
>Port 4: Open
>Port 5: Attempting to use as digital input

Thanks for the quick response!

Keith

**Reply by MAKOMO** [reply]
For whatever reason you have an input filter set to replace received values outside of the specified range 50 to 1000. The HUBs digital ports 0 or 1 results are outside of this range and are thus replaced by the error value -1.

<img width="705" alt="Screenshot 2024-04-22 at 16 52 50" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/8735198f-2454-4cb7-8a1b-875f85107cf7">

Just deactivate that Input Filter.

**Reply by KT-19903** [reply]
That was it! Of course it was something simple. I really appreciate your help troubleshooting this. I had some issues early on configuring my temp probe and clicked on too many things... 

Thank you so much for all your work on this tool.


-Keith
